pkg/antctl/command_definition.go

```diff
@@ -320,6 +327,78 @@ func respTransformer(obj interface{}) (interface{}, error) {
 	return target, nil
 }
 
+// tableOutputForGetCommands formats the table output for "get" commands.
+func (cd *commandDefinition) tableOutputForGetCommands(obj interface{}, writer io.Writer) error {
+	var list []common.TableOutput
+	if reflect.TypeOf(obj).Kind() == reflect.Slice {
+		s := reflect.ValueOf(obj)
+		if s.Len() == 0 || s.Index(0).Interface().(common.TableOutput) == nil {
+			var buffer bytes.Buffer
+			buffer.WriteString("\n")
+			if _, err := io.Copy(writer, &buffer); err != nil {
+				return fmt.Errorf("error when copy output into writer: %w", err)
+			}
+			return nil
+		}
+		for i := 0; i < s.Len(); i++ {
+			ele := s.Index(i)
+			list = append(list, ele.Interface().(common.TableOutput))
+		}
+	} else {
+		ele, _ := obj.(common.TableOutput)
+		list = []common.TableOutput{ele}
+	}
+
+	// Get the elements and headers of table.
+	args := list[0].GetTableHeader()
+	rows := make([][]string, len(list))
+	widths := make([]int, len(args))
+	for i, element := range list {
+		rows[i] = element.GetTableRow(maxTableOutputColumnLength)
+	}
+	sort.Slice(rows, func(i, j int) bool {
+		return rows[i][0] < rows[j][0]
+	})
+	rows = append([][]string{args}, rows...)
```

Comment:
There will be two slice allocation and an unnecessary append operation, especially when the item number is big.
Maybe change to this:
```
        rows := make([][]string, len(list)+1)
        rows[0] = list[0].GetTableHeader()
	for i, element := range list {
		rows[i+1] = element.GetTableRow(maxTableOutputColumnLength)
	}
	sort.Slice(rows[1:], func(i, j int) bool {
		return rows[i][0] < rows[j][0]
	})

	widths := make([]int, len(rows[0]))
        ...
```

Reply:
> There will be two slice allocation and an unnecessary append operation, especially when the item number is big.
> Maybe change to this:
> 
> ```
>         rows := make([][]string, len(list)+1)
>         rows[0] = list[0].GetTableHeader()
> 	for i, element := range list {
> 		rows[i+1] = element.GetTableRow(maxTableOutputColumnLength)
> 	}
> 	sort.Slice(rows[1:], func(i, j int) bool {
> 		return rows[i][0] < rows[j][0]
> 	})
> 
> 	widths := make([]int, len(rows[0]))
>         ...
> ```

This is my original thought on how to solve this problem. However, the code failed to pass some of the unit tests.

Reply:
which unit test and what's the error?

Reply:
> which unit test and what's the error?

First two test cases in command_definition.go. The outputs should be sorted by Name, however, it seems the two outputs just cannot be consistent at the same time. I tried to change the sort order in sortSlice compare function (ascending / descending) but it did not work.

Reply:
I don't understand how the code I suggested would get different output from yours, can you show me the expected output and the actual one?

Reply:
> I don't understand how the code I suggested would get different output from yours, can you show me the expected output and the actual one?

`	// Get the elements and headers of table.
	args := list[0].GetTableHeader()
	rows := make([][]string, len(list)+1)
	rows[0] = args
	widths := make([]int, len(args))
	for i, element := range list {
		rows[i+1] = element.GetTableRow(maxTableOutputColumnLength)
	}
	sort.Slice(rows[1:], func(i, j int) bool {
		return rows[i][0] < rows[j][0]
	})
`



=== RUN   TestCommandList_tableOutputForGetCommands/StructureData-NetworkPolicy-List-HasSummary-RandomFieldOrder
NAME       APPLIED-TO                                       RULES
GroupName1 32ef631b-6817-5a18-86eb-93f4abf0467c             2    
GroupName2 32ef631b-6817-5a18-86eb-93f4abf0467c + 1 more... 1    

    --- PASS: TestCommandList_tableOutputForGetCommands/StructureData-NetworkPolicy-List-HasSummary-RandomFieldOrder (0.00s)
=== RUN   TestCommandList_tableOutputForGetCommands/StructureData-AddressGroup-List-HasSummary-HasEmpty
NAME       POD-IPS                                           
GroupName2 <NONE>                                            
GroupName1 10.0.0.3,127.0.0.1,127.0.0.2,127.0.0.3 + 2 more...

    --- FAIL: TestCommandList_tableOutputForGetCommands/StructureData-AddressGroup-List-HasSummary-HasEmpty (0.00s)
        command_definition_test.go:141: 
            	Error Trace:	command_definition_test.go:141
            	Error:      	Not equal: 
            	            	expected: "NAME       POD-IPS                                           \nGroupName1 10.0.0.3,127.0.0.1,127.0.0.2,127.0.0.3 + 2 more...\nGroupName2 <NONE>                                            \n"
            	            	actual  : "NAME       POD-IPS                                           \nGroupName2 <NONE>                                            \nGroupName1 10.0.0.3,127.0.0.1,127.0.0.2,127.0.0.3 + 2 more...\n"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,4 +1,4 @@
            	            	 NAME       POD-IPS                                           
            	            	+GroupName2 <NONE>                                            
            	            	 GroupName1 10.0.0.3,127.0.0.1,127.0.0.2,127.0.0.3 + 2 more...
            	            	-GroupName2 <NONE>                                            
            	            	 
            	Test:       	TestCommandList_tableOutputForGetCommands/StructureData-AddressGroup-List-HasSummary-HasEmpty


<Click to see difference>

Reply:
It's because the index i and j get other items when sorting, need to add 1 when comparing, i.e. `return rows[i+1][0] < rows[j+1][0]`, but using a sub-slice would be more readable.
```
        rows := make([][]string, len(list)+1)
        rows[0] = list[0].GetTableHeader()
	for i, element := range list {
		rows[i+1] = element.GetTableRow(maxTableOutputColumnLength)
	}
        body := rows[1:]
	sort.Slice(body, func(i, j int) bool {
		return body[i][0] < body[j][0]
	})
```

Reply:
@ZhangYW18 will you update the PR based on the above comment?

Reply:
> @ZhangYW18 will you update the PR based on the above comment?

Done. Sorry for my lateness.